components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -167,270 +235,296 @@ export const SelectTokenModal = React.forwardRef<HTMLDivElement, Props>(
-            token.contractAddress.toLocaleLowerCase() ===
-              searchValue.toLowerCase()
-        )
-      },
-      [getTokensByNetwork, searchValue]
-    )
+    const tokensSortedByValue = React.useMemo(() => {
+      return [...userTokenList].sort(function (a, b) {
+        const aBalance = a.assetBalance
+        const bBalance = b.assetBalance
 
-    const getAccountFiatValue = React.useCallback(
-      (account: BraveWallet.AccountInfo) => {
-        const amounts = getTokensBySearchValue(account).map((token) => {
-          const balance = getBalance(
-            account.accountId,
-            token,
-            tokenBalancesRegistry
-          )
+        const bFiatBalance = computeFiatAmount({
+          spotPriceRegistry,
+          value: bBalance,
+          token: b.asset
+        })
 
-          return computeFiatAmount({
-            spotPriceRegistry,
-            value: balance,
-            token
-          }).format()
+        const aFiatBalance = computeFiatAmount({
+          spotPriceRegistry,
+          value: aBalance,
+          token: a.asset
         })
 
-        const reducedAmounts = amounts.reduce(function (a, b) {
-          return a !== '' && b !== '' ? new Amount(a).plus(b).format() : ''
+        return bFiatBalance.minus(aFiatBalance).toNumber()
+      })
+    }, [userTokenList, spotPriceRegistry])
+
+    const tokensFilteredByNetwork = React.useMemo(() => {
+      if (selectedNetworkFilter.chainId === AllNetworksOption.chainId) {
+        return tokensSortedByValue
+      }
+      return tokensSortedByValue.filter(
+        (token) =>
+          token.asset.chainId === selectedNetworkFilter.chainId &&
+          token.asset.coin === selectedNetworkFilter.coin
+      )
+    }, [tokensSortedByValue, selectedNetworkFilter])
+
+    const tokensFilteredByAccount = React.useMemo(() => {
+      if (selectedAccountFilter.address === AllAccountsOption.address) {
+        return tokensFilteredByNetwork
+      }
+
+      const chainList = filterNetworksForAccount(
+        networks,
+        selectedAccountFilter.accountId
+      ).map((network) => network.chainId)
+
+      return tokensFilteredByNetwork.filter(
+        (token) =>
+          token.asset.coin === selectedAccountFilter.accountId.coin &&
+          chainList.includes(token.asset.chainId)
+      )
+    }, [networks, selectedAccountFilter, tokensFilteredByNetwork])
+
+    const tokensBySearchValue = React.useMemo(() => {
+      if (searchValue === '') {
+        return tokensFilteredByAccount
+      }
+      const lowercaseSearchValue = searchValue.toLowerCase()
+      return tokensFilteredByAccount.filter(
+        (token) =>
+          token.asset.name.toLowerCase() === lowercaseSearchValue ||
+          token.asset.name.toLowerCase().startsWith(lowercaseSearchValue) ||
+          token.asset.symbol.toLocaleLowerCase() === lowercaseSearchValue ||
+          token.asset.symbol.toLowerCase().startsWith(lowercaseSearchValue) ||
```

Comment:
should we replace `.startsWith` with `.includes` to help users find tokens without knowing the exact starting characters in the token name/symbol?

Reply:
Updated to use `.includes`